compiler/GHC/Rename/Expr.hs

```diff
@@ -985,8 +986,8 @@ rebindableContext ctxt = case ctxt of
   ArrowExpr       -> False
   PatGuard {}     -> False
 
-  DoExpr          -> True
-  MDoExpr         -> True
+  DoExpr m        -> isNothing m
```

Comment:
This means that we do not respect `-XRebindableSyntax` within the do expression, right? How does that interact with rebound `ifThenElse` or `fromInteger`?

Reply:
When I did this change, I was conjecturing that it would only affect the do syntax and not the expressions within. I'll double check.

Reply:
Double checked. rebindableContext is only use by Name-lookup functions when constructing renamed statements, it is not used to rename expressions. So looks like `isNothing` is what we want here.